From: Bertrand Lorentz <bertrand.lorentz@gmail.com>
Date: Tue, 3 Apr 2012 20:49:07 +0200
Subject: libossifer: Fix certificate validation with libsoup 2.38 (bgo#673037)

Libsoup 2.38 added a new SOUP_SESSION_SSL_USE_SYSTEM_CA_FILE property,
which tells it to use the default system CA database. If we don't set
it, all SSL validations fail.

Origin: commit:a5c94f2efee425320847d10748410fe9c3ff9311
---
 build/m4/banshee/libwebkit.m4                            | 7 +++++++
 src/Core/Banshee.WebBrowser/libossifer/ossifer-session.c | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/build/m4/banshee/libwebkit.m4 b/build/m4/banshee/libwebkit.m4
index d1f81c8..9a6f7f5 100644
--- a/build/m4/banshee/libwebkit.m4
+++ b/build/m4/banshee/libwebkit.m4
@@ -23,6 +23,13 @@ AC_DEFUN([BANSHEE_CHECK_LIBWEBKIT],
 			AC_DEFINE(HAVE_LIBSOUP_2_38, 1, [libsoup-gnome-2.4 >= 2.38 detected])
 		fi
 
+		PKG_CHECK_MODULES(LIBSOUP_2_38,
+			libsoup-gnome-2.4 >= 2.38,
+			have_libsoup_2_28=yes, have_libsoup_2_28=no)
+		if test x$have_libsoup_2_28 = xyes; then
+			AC_DEFINE(HAVE_LIBSOUP_2_38, 1, [libsoup-gnome-2.4 >= 2.38 detected])
+		fi
+
 		have_libsoup_gnome=no
 		PKG_CHECK_MODULES(LIBSOUP_GNOME,
 			libsoup-gnome-2.4 >= $SOUP_GNOME_MIN_VERSION,
diff --git a/src/Core/Banshee.WebBrowser/libossifer/ossifer-session.c b/src/Core/Banshee.WebBrowser/libossifer/ossifer-session.c
index 64e0215..abf07fb 100644
--- a/src/Core/Banshee.WebBrowser/libossifer/ossifer-session.c
+++ b/src/Core/Banshee.WebBrowser/libossifer/ossifer-session.c
@@ -56,6 +56,12 @@ ossifer_session_initialize (const gchar *cookie_db_path,
                   NULL);
 #endif
 
+#ifdef HAVE_LIBSOUP_2_38
+    g_object_set (session,
+                  SOUP_SESSION_SSL_USE_SYSTEM_CA_FILE, TRUE,
+                  NULL);
+#endif
+
 #ifdef HAVE_LIBSOUP_GNOME
     path = g_strdup_printf ("%s.sqlite", cookie_db_path);
     cookie_jar = soup_cookie_jar_sqlite_new (path, FALSE);
